### ISSUE TITLE
INVESTIGATION: JS Performance: Inspector rerenders

### ISSUE THREAD

**Issue by mhsdesign** [neos/neos-ui#3502]
### Description

Initially i played around with `ClientEval: Math.random()` and wondered why the value actually updates every tick (when used as placeholder). That made me think that there might be useless rerenders.

When i activated the dev tools and enabled the feature to show rerenders, it tells me that the inspector area is always beeing rerendered (Even without my odd use of CLienEval)

Neos 8.3 Neos Ui 8.3 & new Neos Demo (8.3) 

https://github.com/neos/neos-ui/assets/85400359/1367519d-7080-4aad-a420-3a4becc43227

After recording the video i found out that the Neos.Demo:Document.BlogPosting does have some usages ClientEval, so i should probably test this again with simpler elements

[Description of the bug or feature]

### Steps to Reproduce

1. [First Step]
2. [Second Step]
3. [and so on...]

#### Expected behavior

[What you expected to happen]

#### Actual behavior

[What actually happened]

### Affected Versions

Neos:

UI:

You can get this information by running `composer show` or using the package management module within Neos.
The UI version can also be found inside the left sidebar after version 1.0.3.

**Comment by Sebobo** [comment]
Is this still the case, I think I fixed lots of rerenders with #3936

**Comment by mhsdesign** [comment]
Yes i can see that the rerenders do not appear anymore though using

```yaml
inspector:
  editorOptions:
    placeholder: 'ClientEval: Math.random()'
```

as my crazy past self did still causes a rerendering when one refocuses the already selected node e.g. by clicking on it in the structure tree or via the iframe. I guess the check you build in this pr should be `prevProps.focusedNode.contextPath !== this.props.focusedNode.contextPath` or sth?

**Comment by mhsdesign** [comment]
can you read my mind???? you fixed the refocusing problem too ... this seems to be caused by actually reloading the node from the server which causes the js object to be of different identity. 

So https://github.com/neos/neos-ui/pull/3936 closes this fully in combination with https://github.com/neos/neos-ui/issues/4065
